shipyard/src/main/java/org/jclouds/shipyard/handlers/ShipyardErrorHandler.java

```diff
@@ -69,6 +70,12 @@ public void handleError(HttpCommand command, HttpResponse response) {
                if (command.getCurrentRequest().getMethod().equals("POST")) {
                   if (command.getCurrentRequest().getEndpoint().getPath().endsWith("/engines")) {
                      exception = new HttpResponseException("Connection refused registering docker daemon", command, response);
+                  } else if (command.getCurrentRequest().getEndpoint().getPath().endsWith("/containers")) {
+                     exception = new HttpResponseException(command, response, message);
+                  }
+               } else if (command.getCurrentRequest().getMethod().equals("DELETE")) {
+                  if (command.getCurrentRequest().getEndpoint().getPath().endsWith("/servicekeys")) {
+                     exception = new ResourceNotFoundException(message, new HttpResponseException(command, response, message));
```

Comment:
I can make a check to be safe but yes only a 500 with content "service key does not exist" pop on a DELETE at /api/servicekeys